hack/make-rules/go-test/unit.sh

```diff
@@ -61,7 +61,10 @@ fi
 # run unit tests with junit output
 (
   set -x;
+  umask 0022
   mkdir -p "${JUNIT_RESULT_DIR}"
   "${REPO_ROOT}/_bin/gotestsum" --junitfile="${JUNIT_RESULT_DIR}/junit-unit.xml" \
-    -- "./${folder_to_test}"
+    -- \
+    -race \
```

Comment:
@listx it seems there is much more races in the codebase , so we need to fix all of them before enableing it 

https://testgrid.k8s.io/presubmits-test-infra#unit-test&width=20